pkg/cmd/adm/unregister_member_test.go

```diff
@@ -36,9 +39,55 @@ func TestUnregisterMemberWhenAnswerIsY(t *testing.T) {
 	assert.Contains(t, term.Output(), "Delete Member cluster stated above from the Host cluster?")
 	assert.Contains(t, term.Output(), "The deletion of the Toolchain member cluster from the Host cluster has been triggered")
 	assert.NotContains(t, term.Output(), "cool-token")
-
 	AssertDeploymentHasReplicas(t, fakeClient, hostDeploymentName, 1)
-	assert.Equal(t, 2, numberOfUpdateCalls)
+}
+
+func TestUnregisterMemberWhenRestartError(t *testing.T) {
+	// given
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("member-cool-server.com"))
+	hostDeploymentName := test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager")
+	deployment := newDeployment(hostDeploymentName, 1)
+	deployment.Labels = map[string]string{"kubesaw-control-plane": "kubesaw-controller-manager"}
+
+	newClient, fakeClient := NewFakeClients(t, toolchainCluster, deployment)
+	numberOfUpdateCalls := 0
+	fakeClient.MockUpdate = whenDeploymentThenUpdated(t, fakeClient, hostDeploymentName, 1, &numberOfUpdateCalls)
+
+	SetFileConfig(t, Host(), Member())
+	term := NewFakeTerminalWithResponse("y")
+	ctx := clicontext.NewCommandContext(term, newClient)
+
+	// when
+	err := UnregisterMemberCluster(ctx, "member1", func(ctx *clicontext.CommandContext, clusterName string) error {
+		return fmt.Errorf("restart did not happen")
+	})
+
+	// then
+	require.EqualError(t, err, "restart did not happen")
+}
+
+func TestUnregisterMemberCallsRestart(t *testing.T) {
+	// given
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("member-cool-server.com"))
+	hostDeploymentName := test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager")
+	deployment := newDeployment(hostDeploymentName, 1)
+	deployment.Labels = map[string]string{"kubesaw-control-plane": "kubesaw-controller-manager"}
+
+	newClient, fakeClient := NewFakeClients(t, toolchainCluster, deployment)
+	numberOfUpdateCalls := 0
+	fakeClient.MockUpdate = whenDeploymentThenUpdated(t, fakeClient, hostDeploymentName, 1, &numberOfUpdateCalls)
+
+	SetFileConfig(t, Host(), Member())
+	term := NewFakeTerminalWithResponse("y")
+	ctxAct := clicontext.NewCommandContext(term, newClient)
+
+	// when
+	err := UnregisterMemberCluster(ctxAct, "member1", func(ctx *clicontext.CommandContext, restartClusterName string) error {
+		return mockRestart(ctx, restartClusterName)
+	})
+
+	// then
+	require.NoError(t, err)
```

Comment:
I would also explicitly verify that the function was really called:
```suggestion
	called := 0

	// when
	err := UnregisterMemberCluster(ctxAct, "member1", func(ctx *clicontext.CommandContext, restartClusterName string) error {
		called++
		return mockRestart(ctx, restartClusterName)
	})

	// then
	require.NoError(t, err)
	assert.Equal(t, 1, called)
```